crates/matrix-sdk/src/widget/client/actions.rs

```diff
@@ -46,6 +46,9 @@ pub(crate) enum Action {
 pub(crate) struct ReadEventCommand {
     /// Read event(s) of a given type.
     pub(crate) event_type: TimelineEventType,
+    /// If `Some`, read state events with the given key.
+    /// If `None`, read message-like events.
+    pub(crate) state_key: Option<StateKeySelector>,
```

Comment:
Could we make our own enum here? Something like

```rust
enum ReadEventSelector {
  StateEvent { key: String },
  MessageLikeEvents
}
```

It would be more understandable than using an `Option` that means two very different things whether it's set or not.

[edit] Now that I'm seeing that `StateKeySelector` itself is an enum, maybe we could fuse it all together and have a third variant `AnyStateEvent`?

Reply:
Yeah, the current version is closer to the wire format, but that's no good reason to keep it. We can add custom serde code when it comes to that.